File: PengembanganAplikasiMobile/2021-2022--sem2/PAM_DifferentDevices/app/src/main/java/com/septovan/pam/pam_differentdevices/SignUpActivity.java
package com.septovan.pam.pam_differentdevices;

import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.google.android.material.textfield.TextInputLayout;

public class SignUpActivity extends AppCompatActivity {

    private TextInputLayout txtInputSignupUsername;
    private Button btnSignup;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sign_up);

        txtInputSignupUsername = findViewById(R.id.txtInputSignupUsername);
        btnSignup = findViewById(R.id.btnSignup);

        btnSignup.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String username = txtInputSignupUsername.getEditText().getText().toString();
                Intent intentResult = new Intent();
                intentResult.putExtra("username", username);
                setResult(Activity.RESULT_OK, intentResult);
                finish();
            }
        });
    }
}

File: PengembanganAplikasiMobile/2022-2023--sem2/PAM_AcitivityIntent/app/src/main/java/edu/uksw/fti/pam/acitivityintent/SignUpActivity.kt
package edu.uksw.fti.pam.acitivityintent

import android.app.Activity
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.*
import androidx.compose.material.*
import androidx.compose.ui.Modifier
import edu.uksw.fti.pam.acitivityintent.ui.screens.SignUpForm
import edu.uksw.fti.pam.acitivityintent.ui.theme.PAMAcitivityIntentTheme

class SignUpActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            PAMAcitivityIntentTheme {
                // A surface container using the 'background' color from the theme
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = MaterialTheme.colors.background
                ) {
                    SignUpForm(::sendUsernameBackToLogin)
                }
            }
        }
    }
    private fun sendUsernameBackToLogin(username: String?) {
        val result = Intent().putExtra("username", username)
        setResult(Activity.RESULT_OK, result)
        finish()
    }
}